### PR TITLE
Ip 7166 camel

### PR DESCRIPTION
**Pull request by manojkumarbardhan**

### CODE REVIEW

File: docs/_integration-manager/camel-routes/camel-history-management.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+
+## Manage Unprocessed File History
+
+All unprocessed files will be moved to this specified configured location.
+
+Example:
+```
+<route id="manage-unprocessed-file-history" autoStartup="false">
```
Comment (manojkumarbardhan) [inline docs/_integration-manager/camel-routes/camel-history-management.md:29]: Not sure, whether we need to specify these individual properties we have used in this configurations. 
Example - "filterFile" - Why this filter file has been used ? or anything ? Any suggestions?

File: docs/_integration-manager/camel-routes/camel-route-aws.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Example:
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<routes xmlns="http://camel.apache.org/schema/spring">    
+    <route id="run-jobconfig-with-S3-file" autoStartup="false">
+        <setProperty name="jobConfigId">
+            <constant>102712</constant>
+        </setProperty>
+        <setProperty name="filenameOverride">
```
Comment (manojkumarbardhan) [inline docs/_integration-manager/camel-routes/camel-route-aws.md:28]: Not sure, whether we need to specify these individual properties we have used in this configurations. 
Example - "jobConfigId" - What is this value? or anything ? Any suggestions?

File: docs/_integration-manager/camel-routes/camel-route-azure.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+(ProgramDataDirectory)/Actian/FileFolderListener/conf/routes-prototype.xml
+
+NOTE: Multiple routes (i.e. for AWS,Azure, GCP and Local) can be added into a single routes-prototype.xml
+
+Example:
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<routes xmlns="http://camel.apache.org/schema/spring">    
+    <route id="run-jobconfig-with-azure-blob" autoStartup="false">
+        <setProperty name="jobConfigId">
```
Comment (manojkumarbardhan) [inline docs/_integration-manager/camel-routes/camel-route-azure.md:25]: Not sure, whether we need to specify these individual properties we have used in this configurations. 
Example - "jobConfigId" - What is this value? or anything ? Any suggestions?

File: docs/_integration-manager/camel-routes/camel-route-google.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+(ProgramDataDirectory)/Actian/FileFolderListener/conf/routes-prototype.xml
+
+NOTE: Multiple routes (i.e. for AWS,Azure, GCP and Local) can be added into a single routes-prototype.xml
+
+Example:
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<routes xmlns="http://camel.apache.org/schema/spring">    
+    <route id="run-jobconfig-with-gcp-blob" autoStartup="false">
+        <setProperty name="jobConfigId">
```
Comment (manojkumarbardhan) [inline docs/_integration-manager/camel-routes/camel-route-google.md:25]: Not sure, whether we need to specify these individual properties we have used in this configurations. 
Example - "jobConfigId" - What is this value? or anything ? Any suggestions?

This applicable for all type of route configurations.

File: docs/_integration-manager/camel-routes/camel-route-local.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+(ProgramDataDirectory)/Actian/FileFolderListener/conf/routes-prototype.xml
+
+NOTE: Multiple routes (i.e. for AWS,Azure, GCP and Local) can be added into a single routes-prototype.xml
+
+Example:
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<routes xmlns="http://camel.apache.org/schema/spring">    
+    <route id="run-jobconfig-with-local-file" autoStartup="false">
+        <setProperty name="jobConfigId">
```
Comment (manojkumarbardhan) [inline docs/_integration-manager/camel-routes/camel-route-local.md:25]: Not sure, whether we need to specify these individual properties we have used in this configurations. 
Example - "jobConfigId" - What is this value? or anything ? Any suggestions?

This applicable for all type of route configurations.

File: docs/_integration-manager/camel-routes/camel-route-aws.md (hunk 2 above)
Comment (redraiderg) [reply docs/_integration-manager/camel-routes/camel-route-aws.md:28]: Maybe put a README in this folder that van have as much explanation as you need and then we also need to put in a doc bug to get it further fleshed out in there.

Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-aws.md:28]: Thanks Garrett, Yeah we can add but not sure what to add into the README file. And is there any other group we need to add for the PR review review ?@jsnwgnr  - Could you please help with review once & suggest ?

File: docs/_integration-manager/camel-routes/camel-history-management.md (hunk 1 above)
Comment (jsnwgnr) [reply docs/_integration-manager/camel-routes/camel-history-management.md:29]: https://camel.apache.org/components/4.0.x/file-component.html#_endpoint_query_option_filterFile

https://stackoverflow.com/questions/47532169/camel-filterfile-dsl-example

you should explain how filterFile setting works here, namely the camel simple language syntax, e.g. ${file:modified} &lt; ${date:now-14d} means the file was modified over 14 days ago

File: docs/_integration-manager/camel-routes/camel-route-aws.md (hunk 2 above)
Comment (jsnwgnr) [reply docs/_integration-manager/camel-routes/camel-route-aws.md:28]: not sure why we would add a README.md to documentation, but there should be a table with names and descriptions of what the individual properties mean and/or do.

File: docs/_integration-manager/camel-routes/camel-route-azure.md (hunk 3 above)
Comment (jsnwgnr) [reply docs/_integration-manager/camel-routes/camel-route-azure.md:25]: hopefully you know what jobConfigId means, but there should be a table with properties, descriptions, and usage.

File: docs/_integration-manager/camel-routes/camel-route-aws.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+---
+# Camel Route Configuration for AWS S3
+
+## Overview
+
+The File Folder Listner can be configured with Camel route and its typically consists of a series of processing steps that are connected in a linear sequence.
+
+## Route Configuration
+
+Note that the File Folder Listener Service must be restarted for any configuration changes to take effect. Make sure you have already completed: File Folder Listener Authorization
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-route-aws.md]: this is no longer relevant - the service does not need to be restarted for camel route changes

File: docs/_integration-manager/camel-routes/camel-history-management.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+---
+layout: default
+title: Camel File Processing History Management
+nav_order: 5
+parent: File Folder Listener
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-history-management.md]: parent should be Camel Route Configuration, not File Folder Listener

Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-history-management.md]: Done

File: docs/_integration-manager/camel-routes/camel-route-aws.md (hunk 6 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-aws.md]: Done.

File: docs/_integration-manager/camel-routes/camel-route-azure.md (hunk 3 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-azure.md:25]: Done. Please have a look and suggest if any correction needed.

File: docs/_integration-manager/camel-routes/camel-route-aws.md (hunk 2 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-aws.md:28]: Done. Please have a look and suggest if any correction needed.

File: docs/_integration-manager/camel-routes/camel-route-aws.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
+---
+layout: default
+title: Camel Route Configuration for AWS S3
+nav_order: 2
+parent: Camel Route Configuration
+---
+# Camel Route Configuration for AWS S3
+
+## Overview
+
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-route-aws.md:10]: ## Overview

Monitor an S3 folder for new files and submit to the JobConfig run API

File: docs/_integration-manager/camel-routes/camel-route-azure.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
+---
+layout: default
+title: Camel Route Configuration for Azure Blob Storage
+nav_order: 3
+parent: Camel Route Configuration
+---
+# Camel Route Configuration for Azure Blob Storage
+
+## Overview
+
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-route-azure.md:10]: ## Overview

Monitor an Azure Blob Container folder for new files and submit to the JobConfig run API

File: docs/_integration-manager/camel-routes/camel-route-google.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
+---
+layout: default
+title: Camel Route Configuration for Google Cloud Storage
+nav_order: 4
+parent: Camel Route Configuration
+---
+# Camel Route Configuration for Google Cloud Storage
+
+## Overview
+
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-route-google.md:10]: ## Overview

Monitor a GCP Cloud Storage folder for new files and submit to the JobConfig run API

File: docs/_integration-manager/camel-routes/camel-route-local.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
+---
+layout: default
+title: Camel Route Configuration for Local Storage
+nav_order: 1
+parent: Camel Route Configuration
+---
+# Camel Route Configuration for Local Storage
+
+## Overview
+
```
Comment (jsnwgnr) [inline docs/_integration-manager/camel-routes/camel-route-local.md:10]: ## Overview

Monitor a local folder for new files and submit to the JobConfig run API

**Review by jsnwgnr — COMMENTED** [review]
We also need to duplicate these to agent:
docs/_integration_agent/camel-routes/camel-history-management.md
docs/_integration_agent/camel-routes/camel-route-s3.md
etc...

**Comment by manojkumarbardhan** [comment]
> camel-routes

Added for Agent also.

File: docs/_integration-manager/camel-routes/camel-route-local.md (hunk 11 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-local.md:10]: Done.

File: docs/_integration-manager/camel-routes/camel-route-google.md (hunk 10 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-google.md:10]: Done.

File: docs/_integration-manager/camel-routes/camel-route-azure.md (hunk 9 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-azure.md:10]: Done.

File: docs/_integration-manager/camel-routes/camel-route-aws.md (hunk 8 above)
Comment (manojkumarbardhan) [reply docs/_integration-manager/camel-routes/camel-route-aws.md:10]: Done.

**Review by jsnwgnr — APPROVED** [review]

**Review by ntadepalli9 — APPROVED** [review]

**Review by gerardgamb — APPROVED** [review]

**Pull request merged**